tabcmd/commands/auth/session.py

```diff
@@ -353,11 +353,17 @@ def _read_from_json(self):
         if not self._check_json():
             return
         file_path = self._get_file_path()
-        data = {}
+        data = None
         with open(str(file_path), "r") as file_contents:
             data = json.load(file_contents)
+        if not data or data == "" or data == {}:
+            self._wipe_bad_json("File contains no data")
```

Comment:
Excuse my Python inexperience. Can you please help me confirm my understanding?

`open()` returns a file object, so `file_contents` is a file object and not a string containing the contents of the file?

And `json.load()` returns a dictionary? Is the condition to check for null/empty string on 359 necessary?

Reply:
Yes - json.load( ) expects a file object, json.load**s**( ) expects a string. 
You're right that instead of checking for null, I should really be catching JsonDecode errors on json.load. Technically json.load can also return null or an empty string if that's what was in the file - it doesn't require that the content be wrapped in a dictionary. This check also won't catch if it was a regular string.
I cleaned up this and turned it into exception handling, plus added some handling on other file operations, in the updated review.